gssapi/raw/python_gssapi_ext.h

```diff
@@ -1,5 +1,43 @@
 #ifdef OSX_HAS_GSS_FRAMEWORK
 #include <GSS/GSS.h>
+
+// GSS includes these in gssapi_private.h but these aren't present on the host
+// so just explicitly define them.
+GSSAPI_LIB_FUNCTION OM_uint32 GSSAPI_LIB_CALL
+__ApplePrivate_gss_unwrap_iov (
+	OM_uint32 * __nonnull minor_status,
```

Comment:
The nullbility checks appear to have been added to XCode in 6.3, which if I understand the decoder ring correctly, corresponds to 10.9.  At almost five years unsupported, that should be fine as a minimum version :)

However, neither clang nor gcc have support for these (though they appear planned for the upcoming clang-13).  Would this limit our ability to build in certain MacOS setups?  (And, if so, do we care?)

Reply:
That's a good point, I'm not really too well versed in how headers actually work but it sounds like we should just have this as the normal krb5 header to just match out declaration in ext_dce.pyx. Do you see any problems with that?

Reply:
Are you suggesting dropping the extra qualifiers?  If so, I think that'd work - it's the most cautious way to go.  Even without knowing what XCode's compiler does with them, the most we'd lose is some NULL checking - not great, but what can/can't be NULL seems pretty clear for these functions.

But I also don't know the macOS ecosystem well enough to know if building with a non-XCode compiler is a thing people do anymore :)

Reply:
Yea, considering the cython declarations don't have them I'm sure it will work. I'll give it a shot and see how it goes.